crates/walrus-service/src/node.rs

```diff
@@ -2835,53 +2973,100 @@ impl ServiceState for StorageNodeInner {
     async fn store_metadata(
```

Comment:
Remark: I have some trouble following the logic of this function now. It would be helpful if it were split into some dedicated smaller functions or structured with additional comments.